src/partition/repart.c

```diff
@@ -1407,6 +1420,11 @@ static int context_read_definitions(
                 return log_error_errno(r, "Failed to enumerate *.conf files: %m");
 
         STRV_FOREACH(f, files) {
+                const char *drop_in_dir = strjoina(*f, ".d");
```

Comment:
If you read the paragraphs on drop-in units in https://www.freedesktop.org/software/systemd/man/systemd.unit.html, you'll find that drop-ins are not simply searched in the same directory that the main config file is located in, but in all directories that config files are searched in.

So if systemd-repart is looking for configuration files in /usr/lib/repart.d/, etc/repart.d/, and /run/repart.d/, and it finds a config file /etc/repart.d/root-partition.conf, we should then look for drop-in files in /usr/lib/repart.d/root-partition.conf.d/, /etc/repart.d/root-partition.conf.d and /run/repart.d/root-partition.conf.d/.

In other words, for each file, you need to construct a strv of possible dropin directories by iterating over all possible config directories and then pass that list to `conf_files_list_strv()` to get the actual drop-in files.